siliconcompiler/dependency.py

```diff
@@ -0,0 +1,178 @@
+import os
+import requests
+import tarfile
+from pathlib import Path
+import importlib
+from git import Repo, GitCommandError
+from urllib.parse import urlparse
+import shutil
+from time import sleep
+
+
+def path(chip, package, quiet=True):
+    """
+    Compute dependency data path
+    Additionally cache dependency data if possible
+    Parameters:
+    arg1 (package): Package with dependency source info
+        package.dependency (dict):
+            Mandatory keys:
+            'path': '/path/on/network/drive',
+            or
+            'path': 'file:///path/on/network/drive',
+            or
+            'path': e.g. 'git+https://github.com/xyz/xyz/'
+            'ref': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866'
+            or
+            'path': e.g. 'git://github.com/xyz/xyz/'
+            'ref': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866'
+            or
+            'path': e.g. 'git+ssh://github.com/xyz/xyz/'
+            'ref': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866'
+            or
+            'path': e.g. 'ssh://github.com/xyz/xyz/'
+            'ref': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866'
+            or
+            'path': e.g. 'https://github.com/xyz/xyz/archive/',
+            'ref': e.g. '3b94aa80506d25d5388131e9f2ecfcf4025ca866'
+            or
+            'path': e.g. 'https://zeroasic.com/xyz.tar.gz',
+            'ref': e.g. 'your-dependency-version-1'
+    Returns:
+    string: Location of dependency data
+    """
+
+    # Initially try retrieving dependency from schema
+    dependency = {}
+    dependency['path'] = chip.get('dependency', package, 'path')
+    dependency['ref'] = chip.get('dependency', package, 'ref')
+    if not dependency['path']:
+        # If not in the schema retrieve dependency from the python package and store in schema
+        dependency = importlib.import_module(package).dependency
+        for k, v in dependency.items():
+            chip.set('dependency', package, k, v)
```

Comment:
when would this occur?

Reply:
If the dependency is not yet in the schema because pip has not resolved it yet.
Currently we are setting the dependency in the schema directly without using pip's dependency resolution.

Reply:
lets remove this, if the path is blank we should error out